src/datasources/asset/types.ts

```diff
@@ -0,0 +1,140 @@
+import { DataQuery } from '@grafana/schema'
+
+export interface AssetQuery extends DataQuery {
+  type: AssetQueryType,
+  workspace: string,
+  minionIds: string[]
+}
+
+export enum AssetQueryType {
+  Metadata = "Metadata",
+}
+
+export enum EntityType {
+  Asset = "Asset",
+  System = "System"
+}
+
+export interface AssetsResponse {
+  assets: AssetModel[],
+  totalCount: number
+}
+
+export interface AssetModel {
+  modelName: string,
+  modelNumber: number,
+  serialNumber: string,
+  vendorName: string,
+  vendorNumber: number,
+  busType: 'BUILT_IN_SYSTEM' | 'PCI_PXI' | 'USB' | 'GPIB' | 'VXI' | 'SERIAL' | 'TCP_IP' | 'CRIO' | 'SCXI' | 'CDAQ' | 'SWITCH_BLOCK' | 'SCC' | 'FIRE_WIRE' | 'ACCESSORY' | 'CAN' | 'SWITCH_BLOCK_DEVICE' | 'SLSC',
+  name: string,
+  assetType: 'GENERIC' | 'DEVICE_UNDER_TEST' | 'FIXTURE' | 'SYSTEM',
+  firmwareVersion: string,
+  hardwareVersion: string,
+  visaResourceName: string,
+  temperatureSensors: any[], // todo TemperatureSensorModel
+  supportsSelfCalibration: boolean,
+  supportsExternalCalibration: boolean,
+  customCalibrationInterval?: number,
+  selfCalibration?: any, // todo SelfCalibrationModel
+  isNIAsset: boolean,
+  workspace: string,
+  fileIds: string[],
+  supportsSelfTest: boolean,
+  supportsReset: boolean,
+  id: string,
+  location: AssetLocationModel,
+  calibrationStatus: 'OK' | 'APPROACHING_RECOMMENDED_DUE_DATE' | 'PAST_RECOMMENDED_DUE_DATE',
+  isSystemController: boolean,
+  externalCalibration?: ExternalCalibrationModel,
+  discoveryType: 'MANUAL' | 'AUTOMATIC',
+  properties: Record<string, string>,
+  keywords: string[],
+  lastUpdatedTimestamp: string,
+}
+
+export interface AssetPresenceWithSystemConnectionModel {
+  assetPresence: AssetPresence,
+  systemConnection?: SystemConnection
+}
+
+export interface AssetLocationModel {
+  minionId: string,
+  parent: string,
+  resourceUri: string,
+  slotNumber: number,
+  systemName: string,
```

Comment:
Not returned by asset anymore. What is missing from here is physicalLocation that gets populated when the asset is in an unmanaged location (e.g. cabinet).

Reply:
@CiprianAnton did not catch, what exact is not returned anymore?

Reply:
systemName

Reply:
removed, added physicalLocation as string to AssetLocationModel interface